mailing_list/tests.py

```diff
@@ -0,0 +1,242 @@
+        result = mailing_list.sync_listserv_membership()
+
+        mock_add_members.assert_called_with(mailing_list, kwargs['add_members'])
+        mock_delete_members.assert_called_with(mailing_list, kwargs['delete_members'])
+        self.assertEqual(result, kwargs['members_count'])
+
+    def test_sync_listserv_membership_add_members_to_listserv(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu',
+                'sapna_mysore@harvard.edu'
+            },
+            add_members={'david_bonner@harvard.edu', 'sapna_mysore@harvard.edu'},
+            delete_members=set(),
+            members_count=4
+        )
+
+    def test_sync_listserv_membership_delete_members_from_listserv(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu',
+                'sapna_mysore@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu'
+            },
+            add_members=set(),
+            delete_members={'david_bonner@harvard.edu', 'sapna_mysore@harvard.edu'},
+            members_count=2
+        )
+
+    def test_sync_listserv_membership_add_and_delete_members_from_listserv(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'sapna_mysore@harvard.edu'
+            },
+            add_members={'sapna_mysore@harvard.edu'},
+            delete_members={'david_bonner@harvard.edu'},
+            members_count=3
+        )
+
+    @override_settings(IGNORE_WHITELIST=True)
+    def test_sync_listserv_membership_unsubscribed(self):
+        self._call_sync_listserv_membership(
+            listserv_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu'
+            },
+            enrolled_email_set={
+                'hiukei_chow@harvard.edu',
+                'david_downs@harvard.edu',
+                'david_bonner@harvard.edu',
+                'douglas_hall@harvard.edu'
+            },
+            add_members={'david_bonner@harvard.edu'},
+            delete_members=set(),
+            members_count=3
+        )
+
+    def test_sync_listserv_membership_whitelist_not_set(self):
+        self._call_sync_listserv_membership(
```

Comment:
Ok -- in future I think it would be helpful to call some of this stuff out in the test comments, so people can see pretty quickly why the assertion is expected. E.g. a simple `... # should be in whitelist` in the test data would make it a bit clearer just by reading the test. Otherwise I have to go into multiple files to find whether my assumptions about the test's workings are correct. These tests are pretty straightforward and so don't need many comments in general, as the test names, variables, and structure are quite transparent, but when four different tests have identical nearly identical data and assertions, with small tweaks around overrides and fixture checking, it might help to be clearer as to why a given email is or isn't making it into the assertion list. Thanks though, these tests are clear and easy to build on.